jupyterlab_comments/jupyterlab_comments/git_commands.py

```diff
@@ -19,72 +19,70 @@
+        except subprocess.CalledProcessError as e:
+          print("Error invoking git command")
+          print(e.output)
 
-  def status(self, current_path):
-    status = self.run(current_path, 'status').decode('utf-8')
-    return status
+    def status(self, current_path):
+        status = self.run(current_path, 'status').decode('utf-8')
+        return status
 
-  def appraise_pull(self, current_path):
-    self.run(current_path, 'appraise', 'pull', self.remote)
+    def appraise_pull(self, current_path):
+        self.run(current_path, 'appraise', 'pull', self.remote)
 
-  # Return true if the current directory is a git repository
-  def inside_git_repo(self, current_path):
-    try:
-      return_code = subprocess.check_call(['git', 'rev-parse'],
-                                          cwd=current_path)
-      return return_code == 0
-    except Exception as e:
-      print("Error invoking git command")
-      print(e.output)
+    def inside_git_repo(self, current_path):
+        """
+        Return true if the current directory is a git repository.
+        """
+        try:
+          return_code = subprocess.check_call(['git', 'rev-parse'],
+                                              cwd=current_path)
+          print(return_code)
+          return return_code == 0
+        except Exception as e:
+          print("Error invoking git command")
+          print(e.output)
 
-  """
-    Returns a JSON list where each object corresponds to a commment
-    on the given file_path
-    Keys of objects returned: timestamp, author, location, description
-    """
 
-  def get_comments_for_path(self, file_path, current_path):
-    if self.inside_git_repo(current_path):
-      #self.appraise_pull(current_path) #pull new comments from remote repo
-      comments_string = self.run(current_path, 'appraise', 'show', '-d',
-                                 '-json', file_path)
-      comments_json = json.loads(comments_string)
-      comments_list = []
-      if comments_json is not None:
-        for comment_obj in comments_json:
-          comments_list.append(comment_obj['comment'])
-      return comments_list
-    else:
-      #TODO: notify user that they are not connected to a Git repo
-      pass
+    def get_comments_for_path(self, file_path, current_path):
+        """
+        Returns a JSON list of commments for the given file_path.
 
-  def get_code_review_comments(self, file_path, current_path):
-    """
-        1. Use 'git appraise pull' to fetch new comments attached to
-        the current code review
-        2. Use 'git appraise show' to get the comments JSON object
-        3. Parse and return comments
+        Keys of objects returned: timestamp, author, location, description
         """
-    pass
 
-  def add_comment(self, file_path, current_path):
-    pass
+        if self.inside_git_repo(current_path):
+          #self.appraise_pull(current_path) #pull new comments from remote repo
```

Comment:
Sorry, I am just now realizing that the path you use here cannot be the same `file_path` you get from JupyterLab.

Instead, it needs to be the relative path to the file from the root directory of the git repository (which might be a sub-directory of the Jupyter root dir).

You can find the root directory of the git repository using the command `git rev-parse --show-toplevel`.

You can then make the file_path relative to that using `os.path.relpath(file_path, start=git_root_dir)`, where `git_root_dir` is the output of the `git rev-parse...` command above.